core/src/main/java/org/elasticsearch/index/query/Operator.java

```diff
@@ -18,6 +18,7 @@
  */
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.queryparser.classic.QueryParser;
```

Comment:
the lucene enum doesn't have its own ordinals right? so that would mean having our own ordinals anyway, which this existing Operator enum already has. I thought it's also good to use the same Operator enum across all classes that support an operator. Does that make sense?